src/plugins/visualize/public/application/utils/get_top_nav_config.tsx

```diff
@@ -179,6 +181,30 @@ export const getTopNavConfig = (
     stateTransfer.navigateToWithEmbeddablePackage(originatingApp, { state });
   };
 
+  const confirmModal = (
+    <ConfirmModal
+      onConfirm={() => {
+        if (originatingApp) {
+          stateTransfer.navigateToWithEmbeddablePackage(originatingApp);
```

Comment:
Nit: it's unnecessary to use the `navigateToWithEmbeddablePackage` here because we aren't actually passing anything back. You could possibly just call `application.navigateToApp`.